### PR TITLE
Migrating to final PSR-15

### PR DESCRIPTION
**Pull request by moufmouf**
Hey Benoit,

Here is a major PR for phps-cans/psr7-middleware-graphql.
Basically, it migrates the middleware to PSR-15, since PSR-15 now has been adopted for a while.

Also, it targets now `"webonyx/graphql-php": "^0.13"`

I removed anything related to container-interop/service-provider as I believe this belongs in another package.

Finally, I added unit tests (100% test coverage) and Travis / Coveralls tests + PHPStan level 7 + Doctrine coding standard

I think this deserves a new branch. This version should definitely be tagged "0.13.0" (we should follow the version number of "webonyx/graphql-php" in my opinion. Alternatively, we can tag it "1.0.0" and increase minor release for each major webonyx/graphql-php version...

Could you:

- enable Travis on the repository
- merge this in a proper branch and tag this?

Alternatively, if you don't plan to maintain this package, could you give me access rights?
Or I can fork this in a separate package if you prefer. Let me know!

### CODE REVIEW

**Comment by Ngob** [comment]
Hello, thanks for this PR, I will review it asap. I will be glad to receive some help

File: src/WebonyxGraphqlMiddleware.php [hunk 1]
```diff
@@ -1,105 +1,160 @@
...
+        }
+        /*if (strtoupper($request->getMethod()) === 'GET') {
+            $params              = $request->getQueryParams();
+            $params['variables'] = empty($params['variables']) ? null : $params['variables'];
+            $request             = $request->withQueryParams($params);
+        } else {
+            $params              = $request->getParsedBody();
+            $params['variables'] = empty($params['variables']) ? null : $params['variables'];
+            $request             = $request->withParsedBody($params);
+        }*/
```
Comment (Ngob) [inline src/WebonyxGraphqlMiddleware.php]: Unused code

**Comment by Ngob** [comment]
Can you remove the tests part ? I wrote most tests on a separate repository [here](https://github.com/Ngob/psr7-middleware-graphql/blob/master/tests/WebonyxGraphqlMiddlewareTest.php) I do not mind to rewrite them to fit the new PSR

File: src/WebonyxGraphqlMiddleware.php (hunk 1 above)
Comment (moufmouf) [reply src/WebonyxGraphqlMiddleware.php]: Fixed!

**Comment by moufmouf** [comment]
Commented code has been removed!

**Comment by moufmouf** [comment]
Hey Benoît,
Just saw the merge and the tag. I think you made a mistake by tagging a 0.1.1
The new version has a ton of breaking changes.
All users doing a composer update will end up with a non working project.

You should delete tag 0.1.1 and tag a new 0.2 or 1.0 version instead.

**Pull request merged**